From d73f6485678ba1346e94933e797ea2a8e096b72c Mon Sep 17 00:00:00 2001
From: Gleb Kodinets <gkodinets@habana.ai>
Date: Wed, 21 Jul 2021 16:46:26 +0300
Subject: [PATCH] [SW-23529] pci: moving address related defines to embedded specs

This patch moves all address related defines to embedded specs repo
to make them exposed externally.

Change-Id: I08f3d996158d13a070461a9265842eb440b015e8
---

diff --git a/Makefile b/Makefile
index 63ea7bd..26bea84 100644
--- a/Makefile
+++ b/Makefile
@@ -15,6 +15,7 @@
 CFLAGS += -I$(SPECS_DIR)
 CFLAGS += -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)
+CFLAGS += -I$(SPECS_EMBEDDED_DIR)/$(SPECS_EMB_DEV)
 CFLAGS += -include $(SPECS_EMBEDDED_DIR)/$(SPECS_EMB_DEV)/$(SPECS_EMB_DEV)_reg_map.h
 CFLAGS += -Wall -Werror
 CFLAGS += -g -c -fno-builtin
diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index 8088580..d0afa06 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -23,6 +23,7 @@
 #include <crc32.h>
 #include <rma_if.h>	/* required for nvme data */
 #include <hl_nvme_mi_data.h>
+#include <address_space.h>
 
 #include "common.h"
 #include "uart.h"
diff --git a/arch/gaudi/arm/asic.c b/arch/gaudi/arm/asic.c
index 51f4cda..bd8cd01 100644
--- a/arch/gaudi/arm/asic.c
+++ b/arch/gaudi/arm/asic.c
@@ -63,6 +63,7 @@
 #include <boot_info.h>
 #include <cmd_module.h>
 #include <memory.h>
+#include <address_space.h>
 
 #include "common.h"
 
@@ -390,7 +391,7 @@
 	/* Inbound Region 0 - Bar 0 - Point to SRAM + CFG */
 	inbound_region.mode = PCI_BAR_MATCH_MODE;
 	inbound_region.bar = PCI_BAR_ID_0;
-	inbound_region.addr = SDRAM_BASE;
+	inbound_region.addr = SDRAM_FULL_BASE;
 	hl_pci_set_inbound_region(0, &inbound_region);
 
 	/* Inbound Region 1 - Bar 4 - Point to HBM */
@@ -413,8 +414,8 @@
 	inbound_region.bar = PCI_BAR_ID_2;
 	inbound_region.offset_in_bar = sp_sram_offset + HL_SP_SRAM_SIZE;
 	inbound_region.addr = HL_SP_SRAM_BASE + HL_SP_SRAM_SIZE;
-	inbound_region.size = HOST_PHYS_BASE - HL_SP_SRAM_BASE +
-			       HL_SP_SRAM_SIZE;
+	inbound_region.size = HOST_PHYS_BASE -
+			      (HL_SP_SRAM_BASE + HL_SP_SRAM_SIZE);
 	hl_pci_set_inbound_region(3, &inbound_region);
 
 	/* Outbound Region 0 - Point to Host */
diff --git a/arch/gaudi/arm/common.c b/arch/gaudi/arm/common.c
index b5707fc..4be1914 100644
--- a/arch/gaudi/arm/common.c
+++ b/arch/gaudi/arm/common.c
@@ -46,6 +46,7 @@
 #include <security.h>
 #include <hl_boot_desc.h>
 #include <pci.h>
+#include <address_space.h>
 
 #include "common.h"
 
diff --git a/arch/gaudi/arm/common.h b/arch/gaudi/arm/common.h
index 4ec9ec4..635921e 100644
--- a/arch/gaudi/arm/common.h
+++ b/arch/gaudi/arm/common.h
@@ -7,12 +7,10 @@
 #ifndef COMMON_GAUDI_H_
 #define COMMON_GAUDI_H_
 
+#include <address_space.h>
+
 #define PPBOOT_SIZE			0x5000		/* 20KB */
 #define PREBOOT_SIZE			0x20000		/* 128KB */
-#define SRAM_SIZE			0x1400000	/* 20MB */
-
-#define HL_SP_SRAM_BASE			0x7FFBFE0000
-#define HL_SP_SRAM_SIZE			0x10000
 
 #define BOOT_INFO_BASE			HL_SP_SRAM_BASE
 
@@ -22,14 +20,9 @@
 #define PPBOOT_TEXT_BASE		0x7FF0000000
 #define PREBOOT_TEXT_BASE		(PPBOOT_TEXT_BASE + PPBOOT_SIZE)
 
-#define DRAM_BASE			0x0
-#define HL_SPI_BASE			0x7FF8000000ull
 #define HL_SPI_SIZE			0x2000000
 #define HL_CFG_BASE			0x7FFC000000ull
 #define SDRAM_BASE			0x7FF0000000ull
-
-#define HOST_PHYS_BASE			0x8000000000ull
-#define HOST_PHYS_SIZE			0x1000000000000ull
 
 #define FUSE_WORDS_PER_BANK		64
 #define I2C_SLAVE_ADDRESS		0x4f
diff --git a/arch/gaudi/arm/security.c b/arch/gaudi/arm/security.c
index 080363a..eab1142 100644
--- a/arch/gaudi/arm/security.c
+++ b/arch/gaudi/arm/security.c
@@ -68,6 +68,7 @@
 #include <preboot/fit.h>
 #include <memory.h>
 #include <security.h>
+#include <address_space.h>
 
 #include "common.h"
 
diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index 206ce0e..7b43c90 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -8,8 +8,6 @@
 
 #include <bitops.h>
 
-#define SRAM_SIZE			0x3000000	/* 48MB */
-
 #define PPBOOT_SIZE			0x4000		/* 16KB */
 #define PREBOOT_SIZE			0xE000		/* 56KB */
 
diff --git a/arch/gaudi2/arm/asic.c b/arch/gaudi2/arm/asic.c
index 776c8f4..4cde7a2 100644
--- a/arch/gaudi2/arm/asic.c
+++ b/arch/gaudi2/arm/asic.c
@@ -11,8 +11,7 @@
 #include <sha256.h>
 #include <boot_desc.h>
 #include <pci.h>
-
-#include "common.h"
+#include <address_space.h>
 
 u32 hl_skip_pcie_init(void)
 {
diff --git a/arch/gaudi2/arm/common.h b/arch/gaudi2/arm/common.h
index 09c1e83..be13756 100644
--- a/arch/gaudi2/arm/common.h
+++ b/arch/gaudi2/arm/common.h
@@ -32,11 +32,7 @@
 
 #define PPBOOT_SIZE			0x5000		/* 20KB */
 #define PREBOOT_SIZE			0x20000		/* 128KB */
-#define SRAM_SIZE			0x3000000	/* 48MB */
 #define PREBOOT_TEXT_BASE		(PPBOOT_TEXT_BASE + PPBOOT_SIZE)
-
-#define CFG_REGION_SIZE			0xC000000
-#define BAR0_RSRVD_SIZE			0x1000000
 
 #define HL_SPI_BASE			0x3FF6000000ull
 /*
@@ -46,12 +42,6 @@
  * DBG base.
  */
 #define HL_CFG_BASE			0x3FF8000000ull
-#define STM_FLASH_BASE			0x1000007FF4000000ull
-#define BAR0_RSRVD_BASE			0x1000FFFFFC000000ull
-#define DRAM_BASE			0x1001000000000000ull
-
-#define HOST_PHYS_BASE_0		0x0000000000000000ull
-#define HOST_PHYS_SIZE_0		0x0100000000000000ull
 
 #define HL_SP_SRAM_BASE			0x7FF7FE0000ULL
 #define HL_SP_SRAM_SIZE			0x10000
diff --git a/arch/goya/arm/common.h b/arch/goya/arm/common.h
index 6d7981e..e94fd08 100644
--- a/arch/goya/arm/common.h
+++ b/arch/goya/arm/common.h
@@ -7,8 +7,6 @@
 #ifndef COMMON_GOYA_H_
 #define COMMON_GOYA_H_
 
-#define SRAM_SIZE			0x32A0000	/* 50.625MB */
-
 #define PPBOOT_SIZE			0x4000		/* 16KB */
 #define PREBOOT_SIZE			0x20000		/* 128KB */
 
diff --git a/arch/greco/arc/common.c b/arch/greco/arc/common.c
index 2caa856..99963d8 100644
--- a/arch/greco/arc/common.c
+++ b/arch/greco/arc/common.c
@@ -37,7 +37,7 @@
 #include <spi.h>
 #include <preboot/fit.h>
 #include <boot_if.h>
-#include "common.h"
+#include <address_space.h>
 
 #include <dcon0_tstd_masks.h>
 #include <dcon0_tstd_regs.h>
diff --git a/arch/greco/arc/common.h b/arch/greco/arc/common.h
index c3e3e96..fe0ebdf 100644
--- a/arch/greco/arc/common.h
+++ b/arch/greco/arc/common.h
@@ -8,8 +8,6 @@
 
 #include <goya2_blocks_linux_driver.h>
 
-#define SRAM_SIZE			0x8000000	/* 128MB */
-
 #define PPBOOT_SIZE			0x4000		/* 16KB */
 #define PREBOOT_SIZE			0x10400		/* 65KB */
 
@@ -43,11 +41,6 @@
 
 #define RECOVERY_TEXT_BASE	(0x0 + 0x4)
 #define RECOVERY_SIZE		0x4000
-
-#define SDRAM_FULL_BASE		0x7FF0000000ull
-#define DRAM_BASE		0x1000000000000ull
-#define HOST_PHYS_BASE		0x2000000000000ull
-#define HOST_PHYS_SIZE		0x2000000000000ull
 
 #define HL_SPI_BASE		0x1A000000
 #define HL_CFG_BASE		0xFC000000
diff --git a/arch/greco/arc_elf32/common.c b/arch/greco/arc_elf32/common.c
index 06fe083..2873fcb 100644
--- a/arch/greco/arc_elf32/common.c
+++ b/arch/greco/arc_elf32/common.c
@@ -45,8 +45,8 @@
 #include <string.h>
 #include <pci.h>
 #include <boot_if.h>
+#include <address_space.h>
 
-#include "common.h"
 #include <greco.h>
 
 #define PRODUCT_ID_STR_SIZE	0x8
diff --git a/arch/greco/arc_elf32/common.h b/arch/greco/arc_elf32/common.h
index b13c7b6..2dbb617 100644
--- a/arch/greco/arc_elf32/common.h
+++ b/arch/greco/arc_elf32/common.h
@@ -17,11 +17,6 @@
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xC0000
 
 #define SDRAM_BASE		0x10000000
-#define SDRAM_FULL_BASE		0x7FF0000000ull
-#define DRAM_BASE		0x1000000000000ull
-#define HOST_PHYS_BASE		0x2000000000000ull
-#define HOST_PHYS_SIZE		0x2000000000000ull
-
 #define HL_SP_SRAM_BASE		0x7FFBFE0000ULL
 #define PCIE_FW_SRAM_BASE	0x1BFF0000
 #define FW_SRAM_BASE		0x1B050000
diff --git a/arch/greco/arm/asic.c b/arch/greco/arm/asic.c
index 7570783..ece265a 100644
--- a/arch/greco/arm/asic.c
+++ b/arch/greco/arm/asic.c
@@ -16,8 +16,6 @@
 #include <sha256.h>
 #include <boot_desc.h>
 
-#include "common.h"
-
 /* configure yellow LED(GPIO2) in output mode and turn on(active low) */
 void preboot_led_on(void)
 {
diff --git a/arch/greco/arm/common.h b/arch/greco/arm/common.h
index be00d79..d21e2e4 100644
--- a/arch/greco/arm/common.h
+++ b/arch/greco/arm/common.h
@@ -12,7 +12,6 @@
 #define PPBOOT_SIZE			0x5000		/* 20KB */
 #define PREBOOT_SIZE			0x11800		/* 70KB */
 
-#define SRAM_SIZE			0x8000000	/* 128MB */
 #define HL_SP_SRAM_SIZE			0x10000
 
 #define HL_FW_BOOT_PTRS_BASE		0x80000
@@ -31,16 +30,11 @@
 #define HL_SPI_BASE			0x7FF8000000ull
 #define HL_CFG_BASE			0x7FFC000000ull
 #define SDRAM_BASE			0x7FF0000000ull
-#define SDRAM_FULL_BASE			0x7FF0000000ull
-#define DRAM_BASE			0x1000000000000ull
 #define BOOT_ROM_BASE			0x7FFB040000ull
 #define BOOT_ROM_SIZE			0x10000
 #define FW_SRAM_SIZE			0x10000
 
 #define FW_WORD_NUM			0x2000	/* FW_SIZE / 2 */
-
-#define HOST_PHYS_BASE			0x2000000000000ull
-#define HOST_PHYS_SIZE			0x2000000000000ull
 
 #define FUSE_WORDS_PER_BANK		0x40
 #define I2C_SLAVE_ADDRESS		0x6A
diff --git a/arch/greco/common/asic.c b/arch/greco/common/asic.c
index 2c586b4..95fc5a3 100644
--- a/arch/greco/common/asic.c
+++ b/arch/greco/common/asic.c
@@ -15,6 +15,7 @@
 #include <pll.h>
 #include <hl_boot_if.h>
 #include <pci.h>
+#include <address_space.h>
 
 u32 hl_skip_pcie_init(void)
 {
diff --git a/lib/preboot/i2cslave_indirect.c b/lib/preboot/i2cslave_indirect.c
index d070b74..e0b5a1e 100644
--- a/lib/preboot/i2cslave_indirect.c
+++ b/lib/preboot/i2cslave_indirect.c
@@ -9,6 +9,7 @@
 #include <string.h>
 #include <bitops.h>
 #include <preboot.h>
+#include <address_space.h>
 #include <uart.h>
 #include <common/module_i2c_slave.h>
 #include <common/cmd_module.h>
